src/prefect/engine.py

```diff
@@ -1466,6 +1480,23 @@ async def orchestrate_task_run(
                 )
                 args, kwargs = parameters_to_args_kwargs(task.fn, resolved_parameters)
 
+                # update task run name
+                if not run_name_set and task.task_run_name:
+                    sig = inspect.signature(task.fn)
+                    for kwarg, val in sig.parameters.items():
+                        if val.default is not val.empty:
+                            resolved_parameters.setdefault(kwarg, val.default)
```

Comment:
Similarly to https://github.com/PrefectHQ/prefect/pull/8378/files#r1095023065, we shouldn't need this. If we do, let's see if we can find a way to keep this complexity in the callable utils.

Reply:
Same as above - the test failed without these lines